lib/harmony/harmonydb/harmonydb.go

```diff
@@ -0,0 +1,266 @@
+package harmonydb
+
+import (
+	"context"
+	"embed"
+	"errors"
+	"fmt"
+	"math/rand"
+	"regexp"
+	"sort"
+	"strconv"
+	"strings"
+	"time"
+
+	logging "github.com/ipfs/go-log/v2"
+	"github.com/jackc/pgx/v5"
```

Comment:
Since we have pgx v5 and they forked v4, I'm not sure what we are missing. 
But between already supporting fallback and allowing SPs to set the initial server, I don't feel a big need to move to this, especially since only 1 db instance will be common at the beginning.